react-components/src/tools/ManageTools.js

```diff
@@ -0,0 +1,486 @@
+/**
+ * @author aramsey
+ */
+import React, { Fragment, useEffect, useState } from "react";
+
+import ids from "./ids";
+import messages from "./messages";
+import styles from "./styles";
+
+import {
+    build,
+    EmptyTable,
+    EnhancedTableHead,
+    formatMessage,
+    getMessage,
+    LoadingMask,
+    SearchField,
+    TablePaginationActions,
+    withI18N,
+} from "@cyverse-de/ui-lib";
+import {
+    Button,
+    Checkbox,
+    IconButton,
+    Menu,
+    MenuItem,
+    Select,
+    Table,
+    TableBody,
+    TableCell,
+    TablePagination,
+    TableRow,
+    Toolbar,
+    withStyles,
+} from "@material-ui/core";
+import { InfoOutlined, Refresh } from "@material-ui/icons";
+import PropTypes from "prop-types";
+import { injectIntl } from "react-intl";
+
+const TOOL_FILTER_VALUES = {
+    ALL: "ALL",
+    MY_TOOLS: "MY_TOOLS",
+    PUBLIC: "PUBLIC",
+};
+
+const PAGING_OPTIONS = [100, 500, 1000];
+
+function ManageTools(props) {
+    const { parentId, toolList, loading, presenter, intl } = props;
+
+    const [state, setHookState] = useState({
+        selectedTool: null,
+        toolFilterValue: TOOL_FILTER_VALUES.ALL,
+        toolMenuEl: null,
+        shareMenuEl: null,
+        numToolsSelected: 0,
+        order: "asc",
+        orderBy: "name",
+        page: 0,
+        rowsPerPage: PAGING_OPTIONS[0],
+        searchTerm: "",
+    });
```

Comment:
I've tried a few things at this point and still experimenting.  I've tried:
* Caching all menu item functions out into `const onMenuItemClicked` functions instead of having them defined with the elements in the render/return method - I saw articles about how anonymous functions within the render method always cause a re-render
* Did the same with all function calls within the table where I could - the ones that need the specific tool from the `map` function I can't really do much about
* Removed hooks entirely
* Moved the table's map function out of the render method (saw somebody online recommend doing that)
* Moved back to hooks and separated out all the state properties into their own
* Moved out the TablePagination, EnhancedTableHead, and LoadingMask from the `ToolListing` component to have as few props passed to that as possible

Still the same slow performance 😖 Might be time to try https://material-ui.com/components/tables/#virtualized-table

Reply:
I think in the meantime, I'll either push or create a new branch with some/all of these changes I've tried before trying `virtualized-table` so you guys can try it out too if you want.

Reply:
Just for info - how many rows (tools) are there in the table ?

Reply:
I'm getting 433 total in my listing in QA (that includes 8 private tools).

Reply:
OK, I addressed all the PR comments, and decided to go with moving the paging-related stuff to props and then just refresh the listing after add/update/delete for a tool. I also left in a WIP commit that's all the potential optimizations I tried if you guys wanna check it out and give it a go.  Maybe I'm missing something.  I did add in more paging options (25, 50, and 200).  It seems to work fine with 200, but once you hit 500 it's really slow.  Maybe that's not an issue?  I feel like it is though.

Reply:
Thanks! I'll try to test it out today.

I think we would be fine capping the paging at 200~250.

I don't even think our data window "infinite-scroller" shows 500 at a time, though I know it fetches that much at a time from the service; but I think that's just for caching the models, and it displays fewer in the DOM.